tools/autotools/tsk_comparedir.cpp

```diff
@@ -165,18 +165,18 @@ uint8_t
     char fullPath[TSK_CD_BUFSIZE];
     struct stat status;
 
-    strncpy(fullPath, m_lclDir, TSK_CD_BUFSIZE);
+    strncpy(fullPath, m_lclDir, TSK_CD_BUFSIZE - 1);
```

Comment:
It's 2024 and strncpy *still* doesn't guarantee null termination. 
Do you need to add this:
```
fullPath[TSK_CD_BUFSIZE-1]='\0';
```
?

Reply:
(And there are a lot more like this.)

Reply:
I'm going to hold off on the bounded string fixes until later. I've removed those from this PR.